src/coffea/nanoevents/methods/fcc.py

```diff
@@ -0,0 +1,452 @@
+                [[0,1]],
+                []
+            ])
+            >> map_index_to_array(a, a_index)
+                [[44, 33, 23],
+                 [932, 24],
+                 []]
+                ---------------------
+                type: 3 * var * int64
+            >> map_index_to_array(a, a2_index, axis=2)
+                [[[44], [44, 33], [23]],
+                 [[932, 24]],
+                 []]
+                ---------------------------
+                type: 3 * var * var * int64
+
+    """
+    if axis == 1:
+        return array[index]
+    elif axis == 2:
+        axis2_counts_array = awkward.num(index, axis=axis)
+        flat_axis2_counts_array = awkward.flatten(axis2_counts_array, axis=1)
+        flat_index = awkward.flatten(index, axis=axis)
+        trimmed_flat_array = array[flat_index]
+        trimmed_array = awkward.unflatten(
+            trimmed_flat_array, flat_axis2_counts_array, axis=1
+        )
+        return trimmed_array
+    else:
+        raise AttributeError("Only axis = 1 or axis = 2 supported at the moment.")
+
+
+# Function required to create a range array from a begin and end array
+@numba.njit
+def index_range_numba_wrap(begin_end, builder):
+    for ev in begin_end:
+        builder.begin_list()
+        for j in ev:
+            builder.begin_list()
+            for k in range(j[0], j[1]):
+                builder.integer(k)
+            builder.end_list()
+        builder.end_list()
+    return builder
+
+
+def index_range(begin, end):
+    """
+    Function required to create a range array from a begin and end array
+    Example: If,
+            begin = [
+                        [0, 2, 4, 3, ...],
+                        [1, 0, 4, 6, ...]
+                        ...
+                    ]
+            end = [
+                        [1, 2, 5, 5, ...],
+                        [3, 1, 7, 6, ...]
+                        ...
+                    ]
+            then, output is,
+            output = [
+                        [[0], [], [4], [3,4], ...],
+                        [[1,2], [0], [4,5,6], [], ...]
+                        ...
+                    ]
+    """
+    begin_end = awkward.concatenate(
+        (begin[:, :, numpy.newaxis], end[:, :, numpy.newaxis]), axis=2
+    )
+    if awkward.backend(begin) == "typetracer" or awkward.backend(end) == "typetracer":
+        # To make the function dask compatible
+        # here we fake the output of numba wrapper function since
+        # operating on length-zero data returns the wrong layout!
+        # We need the axis 2, therefore, we should return the typetracer layout of [[[]]]
+        awkward.typetracer.length_zero_if_typetracer(
+            begin
+        )  # force touching of the necessary data
+        awkward.typetracer.length_zero_if_typetracer(
+            end
+        )  # force touching of the necessary data
+        return awkward.Array(
+            awkward.Array([[[]]]).layout.to_typetracer(forget_length=True)
```

Comment:
This return value for the TypeTracer does not preserve the contained data type. This could lead to confusion.

Reply:
Yeah. So, a fix might be to have an integer within the braces? eg.
`awkward.Array(awkward.Array([[[0]]]).layout.to_typetracer(forget_length=True))` , so that the contained type is int64 instead of unknown.

Reply:
Hi, sorry for the late update. This snippet does not work. I don't really understand the logic. Could you please explain what `length_one_if_typetracer` does? Alternatively, is there a documentation/example use case for this function that I can refer to? Thanks